include/dfu/mcuboot.h

```diff
@@ -17,26 +17,57 @@
 extern "C" {
 #endif
 
+#ifdef BOOT_SWAP_TYPE_NONE
+#if BOOT_SWAP_TYPE_NONE != 1 /*ensure the same definition in MCUboot */
+#error "definition incompatible"
+#endif
+#else
 /** Attempt to boot the contents of slot 0. */
 #define BOOT_SWAP_TYPE_NONE     1
+#endif
 
+#ifdef BOOT_SWAP_TYPE_TEST
+#if BOOT_SWAP_TYPE_TEST != 2  /*ensure the same definition in MCUboot */
+#error "definition incompatible"
+#endif
+#else
 /** Swap to slot 1.  Absent a confirm command, revert back on next boot. */
 #define BOOT_SWAP_TYPE_TEST     2
+#endif
 
+#ifdef BOOT_SWAP_TYPE_PERM
+#if BOOT_SWAP_TYPE_PERM != 3  /*ensure the same definition in MCUboot */
+#error "definition incompatible"
+#endif
+#else
 /** Swap to slot 1, and permanently switch to booting its contents. */
 #define BOOT_SWAP_TYPE_PERM     3
+#endif
 
+#ifdef BOOT_SWAP_TYPE_REVERT
+#if BOOT_SWAP_TYPE_REVERT != 4  /*ensure the same definition in MCUboot */
+#error "definition incompatible"
+#endif
+#else
 /** Swap back to alternate slot.  A confirm changes this state to NONE. */
 #define BOOT_SWAP_TYPE_REVERT   4
+#endif
 
+#ifdef BOOT_SWAP_TYPE_FAIL
+#if BOOT_SWAP_TYPE_FAIL != 5  /*ensure the same definition in MCUboot */
+#error "definition incompatible"
+#endif
 /** Swap failed because image to be run is not valid */
 #define BOOT_SWAP_TYPE_FAIL     5
+#endif
 
 #define BOOT_IMG_VER_STRLEN_MAX 25  /* 255.255.65535.4294967295\0 */
 
 /* Trailer: */
 #define BOOT_MAX_ALIGN		8
+#ifndef BOOT_MAGIC_SZ
 #define BOOT_MAGIC_SZ		16
+#endif
```

Comment:
`BOOT_MAGIC_SZ` is not needed anymore.

`BOOT_MAX_ALIGN` as well. The only place where you need it is unit test that you have updated. I think MCUBOOT_BOOTUTIL could be linked to unit test to provide that symbol.

Reply:
both definition are needed in mcumgr and dfu/img_utill/flash_img.c under BOOT_TRAILER_IMG_STATUS_OFFS macro.

As you mention these definition can be removed once that code will be linked with MCUBOOT_BOOTUTIL.